### PR TITLE
Fix TLS HTTP hangers in arangobench.

### PR DESCRIPTION
**Pull request by neunhoef**

### CODE REVIEW

**Comment by neunhoef** [comment]
This is a first experiment, but there is a solid theory which could explain the bug. Will post details later.

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7796/

File: 3rdParty/boost/1.71.0/boost/asio/detail/impl/epoll_reactor.ipp [hunk 1]
```diff
@@ -567,10 +573,18 @@ void epoll_reactor::run(long usec, op_queue<operation>& ops)
...
+  // as the epoll_wait call happens. Therefore we are using the official
+  // interrupt() method here, and correspondingly reset() the interrupt when
+  // we have been woken up in epoll_reactor::run.
+#if 0
   epoll_event ev = { 0, { 0 } };
   ev.events = EPOLLIN | EPOLLERR | EPOLLET;
   ev.data.ptr = &interrupter_;
   epoll_ctl(epoll_fd_, EPOLL_CTL_MOD, interrupter_.read_descriptor(), &ev);
+#endif
+  interrupter_.interrupt();
```
Comment (goedderz) [inline 3rdParty/boost/1.71.0/boost/asio/detail/impl/epoll_reactor.ipp]: Maybe make it more explicit like
```cpp
#if 0
...
#else
  interrupter_.interrupt();
#endif
```
or even add a `#define` to base the conditional on, and use it for the other change as well? That way, the patch would be traceable even without having the original code.

Comment (neunhoef) [reply 3rdParty/boost/1.71.0/boost/asio/detail/impl/epoll_reactor.ipp]: Done.

**Pull request merged**